src/entities/TextNode/TextNode.spec.ts

```diff
@@ -0,0 +1,275 @@
+import { describe, it, expect, beforeEach } from '@jest/globals';
+import { TextNode } from './index';
+import { createInlineToolName, FormattingNode } from '../FormattingNode';
+import type { ParentNode } from '../interfaces';
```

Comment:
I think, the `interfaces` folder is redundant. It's better to store all those interfaces inside `src/entities`. For example 

```
src/
  entities/
    ParentNode/
      ParentNode.ts
      ParentNode.spec.ts
```

Reply:
I will refactor the file structure in a separate PR. I want to move all inline entities under the same folder anyway